policybot/pkg/storage/spanner/query.go

```diff
@@ -68,6 +68,74 @@ func (s store) QueryIssuesByRepo(context context.Context, orgLogin string, repoN
 	return err
 }
 
+func (s store) QueryTestResultByName(context context.Context, testName string, cb func(*storage.TestResult) error) error {
+	iter := s.client.Single().Query(context, spanner.Statement{SQL: fmt.Sprintf("SELECT * FROM TestResults WHERE TestName = '%s';", testName)})
+	err := iter.Do(func(row *spanner.Row) error {
+		testResult := &storage.TestResult{}
+		if err := row.ToStruct(testResult); err != nil {
+			return err
+		}
+
+		return cb(testResult)
+	})
+
+	return err
+}
+
+func (s store) QueryTestResultByPrNumber(
+	context context.Context, orgLogin string, repoName string, pullRequestNumber int64, cb func(*storage.TestResult) error) error {
+	sql := `SELECT * from TestResults
+	WHERE OrgLogin = @orgLogin AND 
+	RepoName = @repoName AND 
+	PullRequestNumber = @pullRequestNumber;`
+	stmt := spanner.NewStatement(sql)
+	stmt.Params["orgLogin"] = orgLogin
+	stmt.Params["repoName"] = repoName
+	stmt.Params["pullRequestNumber"] = pullRequestNumber
+	scope.Infof("QueryTestResults SQL\n%v", stmt.SQL)
+
+	iter := s.client.Single().Query(context, stmt)
+	err := iter.Do(func(row *spanner.Row) error {
+		testResult := &storage.TestResult{}
+		if err := row.ToStruct(testResult); err != nil {
+			return err
+		}
+
+		return cb(testResult)
+	})
+
+	return err
+}
+
+func (s store) QueryTestResultByUndone(context context.Context, cb func(*storage.TestResult) error) error {
+	iter := s.client.Single().Query(context, spanner.Statement{SQL: fmt.Sprintf("SELECT * FROM TestResults WHERE Done = false")})
+	err := iter.Do(func(row *spanner.Row) error {
+		testResult := &storage.TestResult{}
+		if err := row.ToStruct(testResult); err != nil {
+			return err
+		}
+
+		return cb(testResult)
+	})
+
+	return err
+}
+
+// Real all rows from table in Spanner and invokes a call back on the row.
+func (s store) QueryAllTestResults(context context.Context, cb func(*storage.TestResult) error) error {
```

Comment:
When is this call actually  used? Can we delete this? Or at least also constrain it to org + repo?